test/index.js

```diff
@@ -233,4 +233,22 @@ test('stop status', function(t) {
   })
 
   mon.start()
-})
\ No newline at end of file
+})
+
+test('restart using restart strategy', function (t) {
+  var mon = respawn([node, crash], { maxRestarts: 3, sleep: [100, 100, 200, 300]})
+  var i = 0;
+  mon.on('spawn', function() {
+    i += 1
+  })
+
+  setTimeout(function() {
+    t.equals(i, 2, "should be on the second restart 200 ms in")
+  }, 200)
+  setTimeout(function() {
+    t.equals(i, 3, "should be on the second restart 300 ms in")
```

Comment:
I don't know if we should look at the maxRestart logic. When I updated the tap dependency to tape for a much improved development experience I had one of the old tests failing; one that delt with the number of restarts.

I don't know if this is a difference between tap and tape--Also, I back ported the unit test to tap when I got it working.